### PR TITLE
Fix System Default Language bug

### LINKED ISSUE #11744: [Bug] Changing language to system default after restart doesn't work
###### Reproduction Steps

1. Change language to anything other than "System default"
2. Restart App
3. Change language to "System Default" (The language preference is properly set to default but the strings and RTL/LTR isn't updated)

###### Expected Result
Language gets set to "System Default"

###### Actual Result
Language isn't set to "System Default"

###### Debug info
Refer to the [support page](https://ankidroid.org/docs/help.html) if you are unsure where to get the "debug info".

```
AnkiDroid Version = 2.16alpha69

Android Version = 11

Manufacturer = unknown

Model = Android SDK built for x86

Hardware = ranchu

Webview User Agent = Mozilla/5.0 (Linux; Android 11; Android SDK built for x86 Build/RSR1.210210.001.A1; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/83.0.4103.106 Mobile Safari/537.36

ACRA UUID = 9743c65f-bcc6-4b19-a5dd-9191b25509e5

Scheduler = std2

Crash Reports Enabled = false

DatabaseV2 Enabled = true

```

###### Research
*Enter an [x] character to confirm the points below:*

- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [x] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### PR DESCRIPTION
**Pull request by dorrin-sot**
## Purpose / Description
Fix bug

## Fixes
Fixes #11744 

## Approach
Set the value for "System Default" to `Resources.getSystem().configuration.locales.get(0)` in `Preferences.GeneralSettingsFragment#initializeLanguageDialog`

## How Has This Been Tested?
Emulator:

https://user-images.githubusercontent.com/59933477/176723160-01b41ad0-5071-43e1-9f68-4eb969ea2648.mov

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [x] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by dorrin-sot** [comment]
I tried 013d069bd8c665534207be68685ce50fc75be530 and 6cbe5f139d60b1620edcfbb4efda6d8d7aebd5db but they also had the issue🤔

**Comment by mikehardy** [comment]
needs reproduction comment answered *and* conflicts (boooo hisss)

**Comment by dorrin-sot** [comment]
I could go as far back as v2.13.0 but the bug is still there

**Comment by david-allison** [comment]
That's unexpected, maybe I'm incorrect

Releases can be downloaded. Try 2.9.6 just in case: https://github.com/ankidroid/Anki-Android/releases/tag/v2.9.6

**Comment by dorrin-sot** [comment]
@david-allison Same on that too :(

**Comment by dorrin-sot** [comment]
Also fixed conflict

**Comment by david-allison** [comment]
Thanks!!

I'm out of office now, please ping me if this isn't reviewed with 24 hours

**Comment by david-allison** [comment]
Thanks!!

I'm out of office now, please ping me if this isn't reviewed with 24 hours

File: AnkiDroid/src/main/java/com/ichi2/anki/preferences/GeneralSettingsFragment.kt [hunk 1]
```diff
@@ -89,7 +90,7 @@ class GeneralSettingsFragment : SettingsFragment() {
         val languageDialogLabels = arrayOfNulls<CharSequence>(items.size + 1)
         val languageDialogValues = arrayOfNulls<CharSequence>(items.size + 1)
         languageDialogLabels[0] = resources.getString(R.string.language_system)
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/preferences/GeneralSettingsFragment.kt:92]: one request to add a `@NeedsTest("")` annotation, code looks good!

**Review by david-allison — APPROVED** [review]

**Comment by lukstbit** [comment]
@dorrin-sot Could you please rebase your changes on top of main? You merged changes from main into your branch and now I can't merge your PR due to that merge commit.

**Comment by david-allison** [comment]
@lukstbit Just to note: the arrow on the `rebase and merge` button in the GitHub UI is still active, and we're able to squash + merge without asking the OP to rebase.

**Comment by dorrin-sot** [comment]
@david-allison Hi sorry I'm abit busy with moving and can't get to the changes you requested in a few days

**Review by mikehardy — APPROVED** [review]
LGTM - I added one trivial commit to fix the lint failure by just moving the NeedsTest import into the lint-approved lexicographically-sorted location

**Comment by dorrin-sot** [comment]
Nice thanks!

**Comment by dorrin-sot** [comment]
BTW Is there a reason there hasn't been any OpenCollective announcement these past few months? Is it cancelled?

**Comment by mikehardy** [comment]
> BTW Is there a reason there hasn't been any OpenCollective announcement these past few months? Is it cancelled?

Hasn't been cancelled, just I'm the only one doing them and I was traveling to visit family (read as: family would kill me if I was on the computer) the entire months of July and August. I'm *still* playing catch up from that. Hoping to get through all the backlog soon, I'll make comments on the PRs

**Comment by dorrin-sot** [comment]
Nice :)

**Pull request merged**